File: java/src/main/java/vsdk/toolkit/raycasting/stochasticRaytracing/StochasticRaytracingPathNode.java
package vsdk.toolkit.raycasting.stochasticRaytracing;

import vsdk.toolkit.common.linealAlgebra.Vector3D;
import vsdk.toolkit.skin.Patch;

/**
Path node: contains all necessary data for computing the score afterwards
*/
public class StochasticRaytracingPathNode {
    public Patch patch;
    public double probability;
    public Vector3D inPoint;
    public Vector3D outpoint;

    public StochasticRaytracingPathNode() {
        patch = null;
        probability = 0.0;
        inPoint = new Vector3D();
        outpoint = new Vector3D();
    }
}

File: typescript/src/main/vsdk/toolkit/raycasting/stochasticRaytracing/StochasticRaytracingPathNode.ts
import { Vector3D } from "../../common/linealAlgebra/Vector3D";
import { Patch } from "../../environment/geometry/elements/Patch";

/**
Path node: contains all necessary data for computing the score afterwards
*/
export class StochasticRaytracingPathNode {
  public patch: Patch | null;
  public probability: number;
  public inPoint: Vector3D;
  public outpoint: Vector3D;

  public constructor() {
    this.patch = null;
    this.probability = 0.0;
    this.inPoint = new Vector3D();
    this.outpoint = new Vector3D();
  }
}
